content/courses/native-onchain-development/program-derived-addresses.md

```diff
@@ -22,258 +22,353 @@ description: "Get a deeper understanding of PDAs."
-previous lessons. This program uses a review creator's public key and the title
-of the movie they're reviewing to find the address that _should_ be used to
-store the review. This approach lets the program create a unique address for
-every new review while also making it easy to locate a review when needed. When
-you want to find a user's review of "Spiderman," you know that it is stored at
-the PDA account whose address can be derived using the user's public key and the
-text "Spiderman" as seeds.
+required mappings into them. It is the natural evolution of the note-taking app
+example we just discussed. If you start to use the note creator's public key as
+a seed to create one map account per user, then why not use both the creator's
+public key and some other known piece of information to derive a PDA for the
+note?
+
+We've been mapping seeds to accounts this entire course and have yet to discuss
+it explicitly. Think about the Movie Review program we've built in previous
+lessons. This program uses a review creator's public key and the title of the
+movie they're reviewing to find the address that _should_ be used to store the
+review. This approach lets the program create a unique address for every new
+review while making it easy to locate a review when needed. When you want to
+find a user's review of "Spiderman", you can derive the PDA account's address
+using the user's public key and the text "Spiderman" as seeds.
 
 ```rust
 let (pda, bump_seed) = Pubkey::find_program_address(&[
         initializer.key.as_ref(),
         title.as_bytes().as_ref()
     ],
-    program_id)
+    program_id
+);
 ```
 
-#### Associated token account addresses
-
-Another practical example of this type of mapping is how associated token
-account (ATA) addresses are determined. Tokens are often held in an ATA whose
-address was derived using a wallet address and the mint address of a specific
-token. The address for an ATA is found using the `get_associated_token_address`
-function which takes a `wallet_address` and `token_mint_address` as inputs.
+### Associated token account addresses
+
+Another practical example of this mapping type is determining the associated
+token account (ATA) addresses. An ATA whose address is derived using a wallet
+address and the mint address of a specific token holds tokens. Solana token
+extensions introduced a new
+[token extensions program ID](https://docs.rs/spl-token-2022/latest/spl_token_2022/fn.id.html).
+You can specify how to create tokens using the legacy token or the new token
+extensions program. Use the new token extensions program, now the standard way
+to create tokens on Solana. The
+[get_associated_token_address_with_program_id](https://docs.rs/spl-associated-token-account/latest/spl_associated_token_account/fn.get_associated_token_address_with_program_id.html)
+function finds the ATA account accepts a `wallet_address`, `token_mint_address`
+and the `token_program_id` as arguments.
```

Comment:
```suggestion
Another practical example of this mapping type is determining associated
token account (ATA) addresses. An ATA is an address used to hold the tokens for a specific account - for example, Jane's USDC account. The ATA address is derived using:

 - the wallet address of the user
 - the mint address of the token
 - the token program used - either the older token program or the newer 
[token extensions program ID](https://docs.rs/spl-token-2022/latest/spl_token_2022/fn.id.html). 

```

I get why it's important to mention token extensions, but let's focus on ATAs for now.

Reply:
The statement is very short and informative. Thanks